beluga_system_tests/test/test_system_new.cpp

```diff
@@ -181,14 +180,19 @@ auto particle_filter_test(
 
     // TODO(nahuel): Update this once the new model features are ready.
     /**
-     * particles |= beluga::actions::propagate(motion(odom)) |
-     *              beluga::actions::reweight(sensor(measurement));
+     * particles |= beluga::actions::propagate(std::execution::par, motion(odom)) |
+     *              beluga::actions::reweight(std::execution::par, sensor(measurement));
      */
     motion.update_motion(odom);
     sensor.update_sensor(std::move(measurement));
-    particles |=
-        beluga::actions::propagate([&motion](const auto& state) { return motion.apply_motion(state, engine); }) |
```

Comment:
Just for a quick test, I tried adding `std::execution::par`, but this code made all the worker threads use the same engine instantiated above, causing the process to hang.

This patch does it the right way, so I figured I better submit this before someone else runs into the same issue.

Reply:
Good find and +1 to the fix, but I think we should make sure this is seldom required. Perhaps turning these PRNGs into thread local singletons and using them implicitly by default i.e. models, distributions, views, etc., would be templated on some engine type, invoking the corresponding singleton when necessary. I don't love it either, but with no control over what execution policies do (and don't) copy and when, the alternatives are way more convoluted.

Reply:
Yeah, currently our random views have default thread_local engine. Doing something similar for models should be enough for the average user. Maybe we don't let users override the engine for motion and sensor models since it's tricky to get right, or we let them do it at their own risk, I'm not which is best.